refactor: treat empty env vars in the same way as not set

kbrabrand opened the pull request:
I find the current handling of `OVSX_PAT`/`VSCE_PAT` a bit confusing. In many/most tools I frequently use an empty value `""` is treated as no value.

After setting up my CI/CD pipe it took quite a bit of time to figure out that the empty value set for `VSCE_PAT` (since we haven't yet added the secret) was treated as a value and not just as not set.

This PR changes proposes to change this. It is a breaking change, but it is with the goal and intention of making the tooling more intuitive. Tests has been added for the set/unset/empty cases.

kbrabrand committed:
fix: treat empty env vars in the same way as if it's not set

Change the handling of OVSX_PAT/VSCE_PAT to treat an empty string in the same way as if it's not
set.

felipecrs commented on lib/utilities.js:
```diff
@@ -7,15 +7,15 @@ function environmentToBoolean(name) {
 
 export function isVscePublishEnabled() {
-  return 'VSCE_PAT' in process.env || isAzureCredentialEnabled();
+  return !!process.env.VSCE_PAT || isAzureCredentialEnabled();
```
Isn't `!!` unnecessary in this case?

kbrabrand replied:
!! is equivalent to doing `Boolean(something)` and is necessary to ensure we get a boolean. Without it a truthy value for `process.env.VSCE_PAT` would mean the function returns a string. The end result is the same, but it's more error prone and harder to reason about.

I've updated the type casts to use `Boolean` for type casting instead, to make the intention clearer.

felipecrs replied:
Thank you, but I changed back to `!!`, which I prefer.

felipecrs reviewed (changes requested):
How about refactoring these too?

https://github.com/felipecrs/semantic-release-vsce/blob/970283712e00a58753e54a363d505ccff2c66b5c/lib/utilities.js#L23

https://github.com/felipecrs/semantic-release-vsce/blob/970283712e00a58753e54a363d505ccff2c66b5c/lib/verify-vsce-auth.js#L9

kbrabrand committed:
refactor: use explicit boolean cast instead of doing !!

kbrabrand committed:
fix: refactor the VSCE_TARGET check

- explicit boolean throw
- remove dead code in `verify-target.js` (code will never get past `if (!isTargetEnabled()) {` if the VSCE_TARGET is cast to a falsy value
- update test to match new behaviour

kbrabrand commented:
> How about refactoring these too?
> 
> https://github.com/felipecrs/semantic-release-vsce/blob/970283712e00a58753e54a363d505ccff2c66b5c/lib/utilities.js#L23
> 
> https://github.com/felipecrs/semantic-release-vsce/blob/970283712e00a58753e54a363d505ccff2c66b5c/lib/verify-vsce-auth.js#L9

I'm fine with doing that. Be aware that it is a breaking change, since it changes the behaviour in cases where the comsumer sets an empty string. But it tracks with my intention of this PR, so I've made the change in my pr.

felipecrs committed:
Simplify to !! again

felipecrs committed:
Remove useless in process.env check

felipecrs committed:
Refactor VSCE_TARGET empty test

felipecrs committed:
Merge branch 'master' into fix/handling-of-empty-env-vars

felipecrs commented:
It's extremely unlikely that this _breaking change_ will hit anyone, and therefore I'll not mark it as a breaking change.

felipecrs commented:
Thanks a lot for your contribution!
